java/pattern/constraint/TypeConstraint.java

```diff
@@ -399,46 +408,85 @@ public int hashCode() {
 
     public static class Owns extends TypeConstraint {
 
+        private static final Set<Annotation> VALID_ANNOTATIONS = set(Annotation.KEY, Annotation.UNIQUE);
+
         private final TypeVariable attributeType;
         private final TypeVariable overriddenAttributeType;
-        private final boolean isKey;
+        private final List<Annotation> annotations;
+        private final Annotation uniqueness;
```

Comment:
We don't want to group annotations into super-types of annotations in the language for validation/restriction purposes.